app/controllers/pending_loans_controller.rb

```diff
@@ -50,39 +55,22 @@ def update_cart
       update_existing_cart
     else
       session[:pending_loan] = {
-        pending_loan_params[:project_id] =>
-          pending_loan_params[:loan_amount]
+        pending_loan_params[:project_id] => pending_loan_params[:loan_amount]
       }
     end
   end
 
   def update_existing_cart
-    if loan_is_already_pending
-      update_specific_loan
-    else
-      session[:pending_loan][pending_loan_params[:project_id]] =
-      pending_loan_params[:loan_amount].to_s
-    end
-  end
-
-  def loan_is_already_pending
-    session[:pending_loan][pending_loan_params[:project_id]]
-  end
-
-  def update_specific_loan
     session[:pending_loan][pending_loan_params[:project_id]] =
-    session[:pending_loan][pending_loan_params[:project_id]]
-  end
-
-  def pending_loans_exist?
-    session[:pending_loan] != nil
+      pending_loan_params[:loan_amount].to_s
   end
 
-  def delete_all_projects_from_cart
-    session.delete(:pending_cart)
+  def delete_project_from_cart
+    session[:pending_loan].delete(pending_loan_params[:project_id])
   end
 
-  def delete_specific_project_from_cart
-    session[:pending_loan].delete(pending_loan_params[:project_id])
+  def valid_loan_amount(max_amount)
+    amount = pending_loan_params[:loan_dollar_amount].to_f
```

Comment:
Yes, all the storing of pending loans in the sessions includes messiness. This functions as expected, but it would definitely be worthwhile to refactor at some point.